enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,340 @@
+
+```console
+Name:         ip-10-0-12-194.ec2.internal
+Namespace:    
+Labels:       <none>
+Annotations:  <none>
+API Version:  machineconfiguration.openshift.io/v1alpha1
+Kind:         MachineConfigNode
+Metadata:
+  Creation Timestamp:  2023-10-17T13:08:58Z
+  Generation:          1
+  Resource Version:    49443
+  UID:                 4bd758ab-2187-413c-ac42-882e61761b1d
+Spec:
+  Node Ref:
+    API Version:  
+    Kind:         Node
+    Name:         ip-10-0-12-194.ec2.internal
+  Pool:
+    API Version:  machineconfiguration.openshift.io/v1
+    Kind:         MachineConfigPool
+    Name:         master
+Status:
+  Conditions:
+    Last Transition Time:  2023-10-17T13:09:02Z
+    Message:               Node has completed update to config rendered-master-cf99e619747ab19165f11e3546c71f1e
+    Reason:                NodeUpgradeComplete
+    Status:                True
+    Type:                  Updated
+    Last Transition Time:  2023-10-17T13:09:02Z
+    Message:               This node has not yet entered the UpdatePreparing phase
+    Reason:                NotYetOccured
+    Status:                False
+  Config Version:
+    Current:            rendered-worker-823ff8dc2b33bf444709ed7cd2b9855b
+    Desired:            rendered-worker-823ff8dc2b33bf444709ed7cd2b9855b
+  Health:               Healthy
+  Most Recent Error:    
+  Observed Generation:  3
+```
+
+The above struct gives us some helpful information about a node as it pertains to the MCO. First, all upgrade related events that have occured on the node from the most recent uprade process (no matter how small) appear in the `conditions`. Other than that, you can see the current and desired machineconfig indicating whether or not the node should be updating as well as whether or not the currently tracked update process held in `conditions` is updating to the expected MachineConfig. The Node refernece and ObservedGeneration exist to let the user know some extra information on the object and how many times we have gone throgh some upgrade related changes.
+
+The states to be reported by this MachineConfigNode will roughly fall into the following:
+
+#### Preparing phase
+ - stopping config drift monitor
+ - reconciling configs
+#### Actionable phase
+- Draining
+- Updating on disk state
+- Updating OS
+#### Post update action
+- Rebooting
+- Closing daemon
+- Node Reboot **OR** Reloading crio
+
+This loops until we are inDesiredConfig
+
+More Paths that will update the node state:
+
+#### Completing Update -- the daemon only gets into this phase (outside of hypershift) when we are inDesiredConfig,it has to do with the drainer annotation
+- Uncordoning
+- Uncordoned
+- Updating node state and metrics
+   		 
+#### Resume phase
+- Start config drift monitor
+- Error states:
+   - update stuck -- specifically degraded and/or failures in the functions this is the phase that happens if we either error in any of the above stages
```

Comment:
I am assuming that any state that will block `Uncordoning` of node will be informed through error states as well?